### ISSUE TITLE
Concerns on logic inside API endpoints which are fetching data

### ISSUE THREAD

**Issue by AndrewThien** [Health-Informatics-UoN/Carrot-Mapper#795]

**Comment by AndyRae** [comment]
Yes I agree - this is a bit of debt we were basically fine with introducing as it works, is necessary for Next to build, and the early stages of the Next work had other priorities. This is still quite low priority though, and shouldn't be much work to undo.
